### PR TITLE
test: add UI interaction tests for missing coverage branches

### LINKED ISSUE #577: Enforce Storybook Test Coverage Thresholds in CI
**Context:**

Our current frontend Storybook testing suite is achieving elite coverage levels (currently ~97% statement/line coverage). To prevent future regressions and ensure all new UI components are rigorously tested, we need to mathematically enforce these coverage minimums in our CI pipeline.

Before locking in the exact numbers, we want to grab any "easy wins" to bump the coverage slightly higher, while explicitly ignoring deep edge cases that aren't suited for UI interaction tests.

**Tasks:**

**Step 1: Pragmatic Coverage Bump (The "Easy Wins")**

* Run `npm run storybook:test:coverage` to generate the latest report.

* Review the uncovered lines. If an uncovered line represents a simple UI interaction state (e.g., clicking a button, expanding a row, rendering a standard state), write the missing Storybook interaction test to cover it.

* **CRITICAL GUARDRAIL:** Do not write Storybook tests for defensive `throw Error` blocks, extreme edge cases, or pure logic functions that are difficult to reach via the UI. We do not want to duplicate the coverage already handled by our standard unit tests. Skip anything that requires complex mocking.

**Step 2: Lock in the Thresholds**

- Once the easy wins are implemented, run `npm run storybook:test:coverage` one final time to establish the new absolute baseline numbers.

- Update our testing configuration (e.g., `vite.config.js` / `vite.config.ts`) to enforce these exact numbers.

Add the thresholds block inside the `test.coverage` object. For example:

JavaScript
```
test: {
  coverage: {
    thresholds: {
      lines: [FINAL_NUMBER],
      branches: [FINAL_NUMBER],
      functions: [FINAL_NUMBER],
      statements: [FINAL_NUMBER]
    }
  }
}
```

**Step 3: Verify CI Failure State**

- Ensure that if coverage drops below these specified thresholds, the test runner exits with a non-zero status code so that the GitHub Actions CI workflow will appropriately fail the build.

### PR DESCRIPTION
**Pull request by markafitzgerald1**
This PR addresses Issue #577 by adding simple UI interaction state tests in Storybook to cover missing code paths. Specifically:
- `Modal.tsx`: Covered the `show=false` path by adding a `Hidden` story.
- `SortOrderInput.tsx`: Covered the `onChange` branch by adding a `ChangeSortOrder` interaction story.
- `Trainer.tsx`: Covered the `initialCards` initialization path by adding a `WithInitialCards` story.

After adding these tests, `npm run storybook:test:coverage` confirmed increased coverage. The thresholds in `vite.config.js` have been updated and locked in to reflect this new baseline. Finally, learnings regarding how to test using `@storybook/test` and `play` functions were documented as a new skill under `skills/testing-storybook`.

---
*PR created automatically by Jules for task [5522654845002356072](https://jules.google.com/task/5522654845002356072) started by @markafitzgerald1*

### CODE REVIEW

File: src/ui-react/Trainer.stories.tsx [hunk 1]
```diff
@@ -161,3 +162,19 @@ export const SortHandInDealOrder = {
...
+    initialCards: [
+      { rank: 0, suit: Suit.SPADES },
+      { rank: 1, suit: Suit.HEARTS },
+      { rank: 2, suit: Suit.DIAMONDS },
+      { rank: 3, suit: Suit.CLUBS },
+      { rank: 4, suit: Suit.SPADES },
+      { rank: 5, suit: Suit.HEARTS },
+    ],
+  },
+  play: async ({ canvasElement }) => {
```
[the remark this answers is not in the document]
Comment (markafitzgerald1) [reply src/ui-react/Trainer.stories.tsx]: Codex: Addressed. The `WithInitialCards` play context is explicitly typed as `{ canvasElement: HTMLElement }`, and `npm run lint` plus the full Docker validation pass both complete successfully.

File: skills/testing-storybook/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+name: storybook-interaction-testing
+description: Guidance on using Playwright in Storybook test components for interaction testing.
+compatibility: Requires @storybook/react-vite and storybook/test.
+---
+
+# Storybook Interaction Testing
+
+**Description:**
+When creating interaction tests in Storybook (e.g., in `*.stories.tsx` or `*.stories.ts` files), ensure that you utilize the proper testing library tools provided by `@storybook/test` or `storybook/test`.
```
[the remark this answers is not in the document]
Comment (markafitzgerald1) [reply skills/testing-storybook/SKILL.md]: Codex: Addressed. The skill markdown is now wrapped and spaced for the repository markdownlint rules, and `npm run lint` passes.

File: patch_tsc.diff [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<<<<<<< SEARCH
```
[the remark this answers is not in the document]
Comment (markafitzgerald1) [reply patch_tsc.diff]: Codex: Addressed. No root-level `patch*.diff` artifacts remain in the PR branch.

File: src/ui-react/Trainer.stories.tsx [hunk 4]
```diff
@@ -161,3 +162,19 @@ export const SortHandInDealOrder = {
 export const SortHandInAscendingOrder = {
   play: createPlay(SortOrder.Ascending),
 };
+
+export const WithInitialCards = {
+  args: {
+    initialCards: [
+      { rank: 0, suit: Suit.SPADES },
+      { rank: 1, suit: Suit.HEARTS },
```
[the remark this answers is not in the document]
Comment (markafitzgerald1) [reply src/ui-react/Trainer.stories.tsx]: Codex: Addressed. The `WithInitialCards` fixture now uses `createCard(...)`, so each story card has the complete runtime `Card` shape, including `count` and `rankLabel`.

File: src/ui-react/Trainer.stories.tsx [hunk 5]
```diff
@@ -161,3 +162,19 @@ export const SortHandInDealOrder = {
...
+      { rank: 2, suit: Suit.DIAMONDS },
+      { rank: 3, suit: Suit.CLUBS },
+      { rank: 4, suit: Suit.SPADES },
+      { rank: 5, suit: Suit.HEARTS },
+    ],
+  },
+  play: async ({ canvasElement }: { canvasElement: HTMLElement }) => {
+    await expect(canvasElement).toHaveTextContent("Hand");
```
[the remark this answers is not in the document]
Comment (markafitzgerald1) [reply src/ui-react/Trainer.stories.tsx]: Codex: Addressed. The story now asserts the rendered hand-card labels from the checkbox labels instead of generic page text, so it verifies the seeded initial hand.

File: skills/testing-storybook/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+name: storybook-interaction-testing
+description: Guidance on using Playwright in Storybook test components for interaction testing.
+compatibility: Requires @storybook/react-vite and storybook/test.
+---
+
+# Storybook Interaction Testing
+
+**Description:**
+When creating interaction tests in Storybook, ensure that you utilize the proper
+testing library tools provided by `@storybook/test` or `storybook/test`.
```
[the remark this answers is not in the document]
Comment (markafitzgerald1) [reply skills/testing-storybook/SKILL.md]: Codex: Addressed. The skill guidance now refers only to this repo's available `storybook/test` import.

File: src/ui-react/Trainer.stories.tsx [hunk 7]
```diff
@@ -161,3 +162,20 @@ export const SortHandInDealOrder = {
...
+      { count: 2, rank: 1, rankLabel: "2", suit: Suit.HEARTS },
+      { count: 3, rank: 2, rankLabel: "3", suit: Suit.DIAMONDS },
+      { count: 4, rank: 3, rankLabel: "4", suit: Suit.CLUBS },
+      { count: 5, rank: 4, rankLabel: "5", suit: Suit.SPADES },
+      { count: 6, rank: 5, rankLabel: "6", suit: Suit.HEARTS },
+    ],
+  },
+  play: async ({ canvasElement }: { canvasElement: HTMLElement }) => {
+    await expect(canvasElement).toHaveTextContent("A");
+    await expect(canvasElement).toHaveTextContent("2");
```
[the remark this answers is not in the document]
Comment (markafitzgerald1) [reply src/ui-react/Trainer.stories.tsx]: Codex: Addressed in commit `452ec51`. The assertion now checks the exact visible seeded hand labels (`6♥`, `5♠`, `4♣`, `3♦`, `2♥`, `A♠`), which fails if `initialCards` is ignored and the default seeded deal is rendered instead.

**Review by markafitzgerald1 — COMMENTED** [review]
lgtm - approved! :)
